src/components/storage-upload-helper.js

```diff
@@ -0,0 +1,49 @@
+import { IMAGE_DOWN_SCALING_WIDTH } from '../domain/constants.js';
+
+const getScaledDimensions = (img, width) => {
+  const ratio = img.naturalWidth / width;
+  return { width, height: Math.round(((img.naturalHeight / ratio) + Number.EPSILON) * 100) / 100 };
+};
+
+const isImage = file => file && file.type.startsWith('image');
```

Comment:
We can only process raster images with the canvas, so files with type `image/svg` will result in an invalid file (if its size is > `IMAGE_DOWN_SCALING_WIDTH`).

Reply:
Interesting, I hadn't considered that 🤔 Should we exclude SVGs from processing then?
I'm not sure if there's another way to resize images on the client side.

Reply:
SVGs do not grow in byte size when they grow in pixel size, so they can be left untouched (it will still show the warning if they're smaller than 1200). So, in my opinion we should do a positive test on well-known image types (`image/png`, `image/jpg`, `image/jpeg`, `image/webp`, so basically those that are supported by `canvas.toBlob()`, others will probably not work anyways).

Reply:
🤦‍♀️ of course, you are right, they are vector images. Will do a white-list check!

Reply:
Done, thank you!